### PR TITLE
[go-gen] Fix list DAO request query arguments

### PR DESCRIPTION
**Pull request by willjvsmith**
* Completes the TODO @xsanda found
* Handles the case where a service is `#enumerable` but `#readable(by: all)`, so doesn't need `input.AuthID` in the query arguments for the prepared statement
  * For example: `rows, err := executeQueryWithRowResponses(dao.DB, "SELECT id, created_by, test FROM bar;", input.AuthID)`

I've added a `temple test` integration test for this.

### CODE REVIEW

**Review by c-harding — APPROVED** [review]
Why not add it to the complex E2E test? Updating the tests is trivial now, just a copy from `/tmp/` to `resources/`

**Review by Lewiky — APPROVED** [review]
Ye boi

File: src/test/scala/temple/testfiles/enumReadAll.temple [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Bulbasaur: service {
+  nickname: string;
+  #auth;
+}
+
+Squirtle: service {
+  nickname: string;
+  #enumerable;
+  #readable(by: all);
+}
```
Comment (c-harding) [inline src/test/scala/temple/testfiles/enumReadAll.temple]: trailing newline plz

**Review by c-harding — APPROVED** [review]
lgtm

**Review by jaylees14 — APPROVED** [review]
Just @xsanda's comment, but LGTM

**Pull request merged**